tests/recognizer-tests.js

```diff
@@ -129,6 +129,16 @@ test("A dynamic route recognizes", function() {
   equal(router.recognize("/zoo/baz"), null);
 });
 
+test("A dynamic route with encoded params", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo/:bar", handler: handler }]);
+
+  equal(router.recognize("/foo/bar%20baz")[0].params.bar, "bar baz");
+  equal(router.recognize("/foo/bar%23baz")[0].params.bar, "bar#baz");
+  equal(router.recognize("/foo/bar%3fbaz")[0].params.bar, "bar?baz");
+});
+
```

Comment:
Hmm. `(` and `)` are going to wreak havoc with the current assumption that paths given to `add` are decoded. I would naturally have written
```js
router.add([ { path: '/lol(/(wat))' } ]);
```
but clearly we need a way to escape parentheses. We could call `decodeURIComponent` on the `path` argument to `add`, but how many times? Until it stops changing? Just once? How do you communicate that API change?